mxcubeweb/core/components/harvester.py

```diff
@@ -0,0 +1,505 @@
+            "contents": contents,
+            "global_state": {"global_state": global_state, "commands_state": cmdstate},
+            "cmds": {"cmds": cmds},
+            "msg": msg,
+            "plate_mode": HWR.beamline.diffractometer.in_plate_mode(),
+        }
+
+        return initial_state
+
+    def get_harvester_contents(self):
+        """Get the Harvester contents info
+
+        Return (Dict): Dict content, object name etc..
+        crystal_list and number of available pins"
+        """
+        if HWR.beamline.harvester:
+            root_name = HWR.beamline.harvester.__TYPE__
+            crystal_list = self.get_crystal_list()
+            room_temperature_mode = HWR.beamline.harvester.get_room_temperature_mode()
+            number_of_pins = HWR.beamline.harvester.get_number_of_available_pin()
+            contents = {
+                "name": root_name,
+                "harvester_crystal_list": crystal_list,
+                "number_of_pins": number_of_pins,
+                "calibration_state": self.get_calibration_state(),
+                "room_temperature_mode": room_temperature_mode,
+            }
+        else:
+            contents = {"name": "OFFLINE"}
+
+        return contents
+
+    def get_crystal_list(self):
+        """Get the Harvester Sample List info
+
+        Return (List):  list of dict content
+        state , name etc.. of the current processing plan"
+        """
+        crystal_list = []
+
+        try:
+            harvester_device = HWR.beamline.harvester
+            crystal_uuids = harvester_device.get_crystal_uuids()
+            crystal_names = harvester_device.get_sample_names()
+            crystal_acronymes = harvester_device.get_sample_acronyms()
+            sample_states = harvester_device.get_samples_state()
+            for index, x_tal in enumerate(crystal_uuids):
+                img_url = HWR.beamline.harvester.get_crystal_images_urls(x_tal)
+                img_target_x = HWR.beamline.harvester.get_image_target_x(x_tal)
+                img_target_y = HWR.beamline.harvester.get_image_target_y(x_tal)
+
+                if len(crystal_acronymes) > 0 and len(crystal_acronymes) == len(
+                    crystal_uuids
+                ):
+                    proteinAcronym = crystal_acronymes[index]
+                else:
+                    proteinAcronym = (
+                        crystal_acronymes[0] if len(crystal_acronymes) > 0 else ""
+                    )
+                lst = {
+                    "crystal_uuid": x_tal,
+                    "name": crystal_names[index],
+                    "state": sample_states[index],
+                    "acronym": proteinAcronym,
+                    "img_url": img_url,
+                    "img_target_x": img_target_x,
+                    "img_target_y": img_target_y,
+                }
+                crystal_list.append(lst)
+        except Exception as ex:
+            print("Could not get Crystal List : %s" % str(ex))
```

Comment:
Why no proper logging?

```suggestion
            some_logger.exception("Could not get Crystal List")
```